### ISSUE TITLE
nginx url remaping

### ISSUE THREAD

**Issue by fuinhaazulgit** [OmniDB/OmniDB#833]
Hi there.

I´d like to put omnidb-server behind a url just like: http://myserver.com/omni  with nginx.
I tried a lot of options, but none had good results. Now I have:

server {
 listen 80, 443;
 server_name myserver.com;
 root /servers/myserver;
 index index.jsp index.html index.htm;

 location /omni/ {
  proxy_pass http://localhost:9000/;
  
  sub_filter "'/static" "'/omni/static/";
  sub_filter '"/static' '"/omni/static/';
  sub_filter "'/check_session_message" "'/omni/check_session_message/";
  sub_filter '"/check_session_message' '"/omni/check_session_message/';
  sub_filter "'/sign_in" "'/omni/sign_in/";
  sub_filter '"/sign_in' '"/omni/sign_in/';
  sub_filter_once off;
  sub_filter_types *; 

  proxy_set_header   X-Real-IP $remote_addr;
  proxy_set_header   X-Forwarded-For $proxy_add_x_forwarded_for;
  proxy_set_header   X-Forwarded-Ssl https;
  proxy_set_header   X-Forwarded-Proto $scheme;
  proxy_set_header   X-Forwarded-Port 443;
  proxy_set_header   X-Forwarded-Host $host;
  proxy_set_header   Host $host;
  proxy_http_version 1.1;
  proxy_set_header Upgrade $http_upgrade;
  proxy_set_header Connection "upgrade";
 }

location /workspace/ {
  proxy_pass http://localhost:9000/workspace/;

  sub_filter '"/static' '"/omni/static/';
  sub_filter "'/check_session_message" "'/omni/check_session_message/";
  sub_filter '"/check_session_message' '"/omni/check_session_message/';
  sub_filter "'/sign_in" "'/omni/sign_in/";
  sub_filter '"/sign_in' '"/omni/sign_in/';
  sub_filter_once off;
  sub_filter_types *; 
  
  proxy_set_header   X-Real-IP $remote_addr;
  proxy_set_header   X-Forwarded-For $proxy_add_x_forwarded_for;
  proxy_set_header   X-Forwarded-Ssl https;
  proxy_set_header   X-Forwarded-Proto $scheme;
  proxy_set_header   X-Forwarded-Port 443;
  proxy_set_header   X-Forwarded-Host $host;
  proxy_set_header   Host $host;
  proxy_http_version 1.1;
  proxy_set_header Upgrade $http_upgrade;
  proxy_set_header Connection "upgrade";
 }

 location /wss/ {
  proxy_pass http://localhost:26500/;
  proxy_set_header   X-Real-IP $remote_addr;
  proxy_set_header   X-Forwarded-For $proxy_add_x_forwarded_for;
  proxy_set_header   X-Forwarded-Ssl https;
  proxy_set_header   X-Forwarded-Proto $scheme;
  proxy_set_header   X-Forwarded-Port 443;
  proxy_set_header   X-Forwarded-Host $host;
  proxy_set_header   Host $host;
  proxy_http_version 1.1;
  proxy_set_header Upgrade $http_upgrade;
  proxy_set_header Connection "upgrade";
 }

 
}

There´s an easy way to to this? Im getting crazy with this.

Thanks!!!

**Comment by greenaussie** [comment]
You seem to have some weird quoting and doubling up of filter strings (maybe that is just a cut and past thing).  sub_filter is not a particularly robust solution, and it would be safer for the application to handle a path prefix (I also tried a few experiments, unsuccessfully, inserting [Content-Base](https://tools.ietf.org/html/rfc2068#section-14.11) headers without success). 

I have that broadly working (see below), except for the logout redirection (I suspect sub_filter does not affect headers). I added a path to wss for consistency. Also affected by [this bug which should be resolved in next release](https://github.com/OmniDB/OmniDB/issues/652).

The whole section without snipping is at [this gist](https://gist.github.com/greenaussie/d10f3b34350133dfebf76df2e5141c42). 

 I also tried a few experiments, unsuccessfully, inserting [Content-Base](https://tools.ietf.org/html/rfc2068#section-14.11) headers without success. 

```
        location /omnidb/ {
                proxy_pass http://localhost:8000/ ;
                # add_header Content-Base /omnidb/ ;
                proxy_http_version 1.1;
                proxy_set_header Upgrade $http_upgrade;
                proxy_set_header Connection 'upgrade';
                proxy_set_header Host $host;
                proxy_cache_bypass $http_upgrade;

                sub_filter '/wss' '/omnidb/wss';

                # From https://github.com/OmniDB/OmniDB/OmniDB_app/urls.py

                #LOGIN
                sub_filter '/static' '/omnidb/static';
                sub_filter '/sign_in' '/omnidb/sign_in';
                # sub_filter '/login' '/omnidb/login';
                # sub_filter '/login.css' '/omnidb/login.css';
                sub_filter '/check_session_message' '/omnidb/check_session_message';
                sub_filter '/logout' '/omnidb/logout';

                #CONNECTION
                sub_filter '/connections' '/omnidb/connections';
                sub_filter '/get_connections' '/omnidb/get_connections';
                sub_filter '/save_connections' '/omnidb/save_connections';
                sub_filter '/test_connection' '/omnidb/test_connection';
                sub_filter '/select_connection' '/omnidb/select_connection';

< snip hundreds of lines >

               sub_filter '/update_saved_monitor_unit_interval' '/omnidb/update_saved_monitor_unit_interval';
                sub_filter '/workspace' '/omnidb/workspace';

                sub_filter_once off;
                sub_filter_types *;

        }

        location /omnidb/workspace/ {
                proxy_pass http://localhost:8000/workspace/ ;
                # add_header Content-Base /omnidb/ ;
                proxy_http_version 1.1;
                proxy_set_header Upgrade $http_upgrade;
                proxy_set_header Connection 'upgrade';
                proxy_set_header Host $host;
                proxy_cache_bypass $http_upgrade;

                sub_filter '/static' '/omnidb/static';
                sub_filter '/logout' '/omnidb/logout';
                sub_filter '/wss' '/omnidb/wss';
                sub_filter_once off;
                sub_filter_types *;

        }

    location /omnidb/wss {
        proxy_pass  http://127.0.0.1:25482;
        proxy_set_header   X-Real-IP $remote_addr;
        proxy_set_header   X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header   X-Forwarded-Ssl https;
        proxy_set_header   X-Forwarded-Proto https;
        proxy_set_header   X-Forwarded-Port 443;
        proxy_set_header   Host $host;
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "upgrade";
    }
```

**Comment by bischofs** [comment]
I think just adding a "external domain" or "relative URL" option in the omni.conf like gitlab does it (https://docs.gitlab.com/ee/install/relative_url.html) would be the best option. I suspect nginx can handle it but it looks complicated.

**Comment by wind39** [comment]
Hi @fuinhaazulgit , @greenaussie , @bischofs ,

Today we released OmniDB 2.16.0, so users will be able to host OmniDB server in a custom URL path.

Please try it out. Keeping this issue open for a few more days in case you want to give some feedback.

Thanks in advance,

William

**Comment by wind39** [comment]
Closing this issue now.
